client/components/payment-activity/hooks.ts

```diff
@@ -0,0 +1,132 @@
+/**
+ * External dependencies
+ */
+import { useState } from 'react';
+import { __ } from '@wordpress/i18n';
+import moment from 'moment';
+
+interface DateRange {
+	/** The name of the date range preset. e.g. last_7_days */
+	preset_name: string;
+	/** The date range start datetime used to calculate transaction data, e.g. 2024-04-29T16:19:29 */
+	date_start: string;
+	/** The date range end datetime used to calculate transaction data, e.g. 2024-04-29T16:19:29 */
+	date_end: string;
+}
+
+/**
+ * Hook to manage the selected date range and date range presets for the payment activity widget.
+ */
+export const usePaymentActivityDateRangePresets = (): {
```

Comment:
Because this is a complex date-wrangling piece of logic, it may be good to either test this:
* As an integration test, testing the `PaymentActivity` component to ensure the correct date is selected.
* As an unit test, turning `dateRangePresets` into a function that we can test with different `now` dates

Manual testing via UI is looking good, however!

Reply:
We can log an issue to add tests.

Reply:
Added an issue to track this:
https://github.com/Automattic/woocommerce-payments/issues/8943